src/stores/map.ts

```diff
@@ -156,30 +217,40 @@ export const useMapStore = defineStore("map", () => {
   ) {
     const map = await getMap();
     map.setFeatureState(
-      { source: "points", id: spaceId },
+      { source: "clustered-points", id: spaceId },
       { buildingOccupancy }
     );
   }
 
   // Updates all data for all points
   async function updateData() {
     const map = await getMap();
-    const source = map.getSource("points") as GeoJSONSource;
-    source.setData(geoJsonSpaces.value);
+    const source = map.getSource("clustered-points") as GeoJSONSource;
+    if (source) {
```

Comment:
Is there a reason why the `if` is needed now?

Reply:
I was getting mapbox warning that there was no source yet for updating the new data with clustered buildings so added this and it worked

Reply:
Could that be linked to the changes in promise handling for uploading markers? Looks a bit weird :)